### ISSUE TITLE
DM tags have regressed.

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#12420]
from https://matrix.to/#/!pcfWjiETvnVuspPLPl:mozilla.org/$wuGewUkYRd3s3jCTp-1WkdPJKNoAS1aAig3cYx9Xk4M?via=mozilla.org&via=matrix.org&via=maunium.net:

 * When I start a new dm it doesn't go into the direct messages section but rather to rooms. On the other side it shows correctly.
 * I can't mark it as a dm either. The menu menu just closes without anything happening.

and from https://matrix.to/#/!pcfWjiETvnVuspPLPl:mozilla.org/$BaCLDqrv8d6qaKv9LPeTqpgeE3C0VLpSSTJis5dWpvE?via=mozilla.org&via=matrix.org&via=maunium.net
 * i have a direct chat with users X and Y.  i want to have a direct chat with just X, but selecting them in the direct messages panel  brings me to the chat with X and Y (on riot web).  that seems.. wrong?

**Comment by turt2live** [comment]
it's possible this is related to the rate limiting on the room lists update and not a true regression.

**Comment by ara4n** [comment]
the first issue sounds like that.
the second issue sounds... weird

**Comment by turt2live** [comment]
oh, the second one is a cry for help and the definition of why we need canonical DMs.

**Comment by ara4n** [comment]
why would selecting X in the DMs panel ever take you to a chat with X and Y?  It should have been labelled X and Y.

**Comment by turt2live** [comment]
The current DM map maps one user to many rooms, not necessarily one room to many users.

**Comment by turt2live** [comment]
It's certainly a bug, but the root cause is that the DM map is poorly calculated and ends up with these sorts of bugs.

**Comment by Uumas** [comment]
The first issue gets fixed by clearing Riot cache

**Comment by mcomella** [comment]
I have experienced this when using the desktop client on macOS: I created a DM and it was categorized as a Room. After this, when I opened the web interface (chat.mozilla.org), several additional DMs were also present in Rooms. I was able to categorize those DMs as DMs in the web interface but unable to make similar changes in the desktop client; the changes I made in the web interface aren't reflected in the desktop client either.

---

> The first issue gets fixed by clearing Riot cache

How do I clear the Riot cache?
